File: app/src/main/java/com/roy/ui/ABase.java
package com.roy.ui;

import android.app.ActivityManager;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;

import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import com.roy.R;
import com.roy.util.UtilSettings;

public class ABase extends AppCompatActivity {

    protected UtilSettings mUtilSettings;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        mUtilSettings = new UtilSettings(this);
        if (savedInstanceState == null) {
            updateNightMode();
        }
        super.onCreate(savedInstanceState);
    }

    @Override
    public void setContentView(@LayoutRes int layoutResID) {
        super.setContentView(layoutResID);
        setTaskDescription();
    }

    private void setTaskDescription() {
        Bitmap appIconBitmap = BitmapFactory.decodeResource(getResources(), R.mipmap.ic_launcher);
        ActivityManager.TaskDescription taskDescription = new ActivityManager.TaskDescription(
                getString(R.string.app_name),
                appIconBitmap,
                ContextCompat.getColor(getBaseContext(), R.color.colorPrimaryDark));
        setTaskDescription(taskDescription);
    }

    protected void updateNightMode() {
        if (mUtilSettings == null) {
            mUtilSettings = new UtilSettings(this);
        }
        getDelegate().setLocalNightMode(mUtilSettings.getNightMode());
    }
}

File: app/src/main/java/com/mckimquyen/ui/ABase.kt
package com.mckimquyen.ui

import android.app.ActivityManager.TaskDescription
import android.graphics.BitmapFactory
import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.mckimquyen.R
import com.mckimquyen.util.UtilSettings

open class ABase : AppCompatActivity() {
    @JvmField
    protected var utilSettings: UtilSettings? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        utilSettings = UtilSettings(this)
        if (savedInstanceState == null) {
            updateNightMode()
        }
        super.onCreate(savedInstanceState)
    }

    override fun setContentView(@LayoutRes layoutResID: Int) {
        super.setContentView(layoutResID)
        setTaskDescription()
    }

    private fun setTaskDescription() {
        val appIconBitmap = BitmapFactory.decodeResource(resources, R.mipmap.ic_launcher)
        val taskDescription = TaskDescription(
            getString(/* resId = */ R.string.app_name),
            appIconBitmap,
            ContextCompat.getColor(baseContext, R.color.colorPrimaryDark)
        )
        setTaskDescription(taskDescription)
    }

    protected fun updateNightMode() {
        if (utilSettings == null) {
            utilSettings = UtilSettings(this)
        }
        utilSettings?.let {
            delegate.localNightMode = it.nightMode
        }
    }
}